lib/Log/OwncloudLog.php

```diff
@@ -109,14 +129,39 @@ public function write($type, $diagnostics) {
 		}
 
 		$entry = json_encode($entry);
+		$this->logsQueue[] = array('entry' => $entry, 'timestamp' => $timestamp);
```

Comment:
Hm ... could you clarify how to enable diagnosis only for certain queries. The entries are stored in memory now. Which *will* cause out of memory errors when syncing large ldap directories or running file scans on huge directories. Can you add an option not to sort the entries? Should be done subsequently with running sort over the logfile anyway. The log file is sortable via cli, isn't it? ... hm no it isn't. Please reorder the fields so time is logged first. This may require changing the format for compact entries.